app/helpers/i18n_helper.rb

```diff
@@ -47,6 +47,40 @@ def date_eras_gregorian(date)
     nil
   end
 
+  # Return the most accurate date from an array of dates using scoring algorithm below
+  def date_most_accurate(dates)
+    return nil unless dates.present? && (dates.is_a?(Array) || dates.is_a?(String))
+    dates = [dates] if dates.is_a?(String)
+    most_accurate = nil
```

Comment:
assign the first gregorian date here already to avoid the extra conditional block in the each loop